### PR TITLE
setting `visible=False` to `gr.Group` hides it

### PR DESCRIPTION
**Pull request by abidlabs**
Closes: #4505

In order to make sure this regression does not happen again, I wrote a TS test. However, I'm not happy with this test, as it tests implementation (the presence of a class name) instead of behavior (checking to see if the CSS display is actually set to "none"). 

I tried adding these lines of JS to the test but it wouldn't work for me:

at the beginning of the test:

```js
// Wait for any potential reactivity updates in the component
await waitFor(() => {});
```

and at the end of the test:

```js
const computedStyles = window.getComputedStyle(groupElement);
assert(computedStyles.display === 'none', "Group element is not hidden.");
```

If we can get this to work, that'd be much better!

### CODE REVIEW

File: js/app/src/components/Group/Group.svelte [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	export let visible: boolean = true;
 </script>
 
-<div id={elem_id} class={elem_classes.join(" ")} class:hidden={!visible}>
+<div id={elem_id} class={elem_classes.join(" ")} class:hide={!visible}>
```
Comment (abidlabs) [inline js/app/src/components/Group/Group.svelte:7]: Renaming to be consistent with `gr.Column`, `gr.Row`, etc.

**Comment by pngwn** [comment]
I'd expect the code you posted to work without any need to await reactive changes because it should all happen synchronously. Will take a look.

**Comment by abidlabs** [comment]
Thanks @pngwn that's what I had thought as well!

**Comment by pngwn** [comment]
Okay, I understand it now. These virtual DOM environments that we use to test in node (`JSDom` `happy-dom`) don't fully implement `window.getComputedStyles()`. I'm not sure about `JSDom` but `happy-dom` seems to do some half-way thing (that I think it confusing). If the style is set directly with the `style` property `<element style="display: none;" />` then it will show up on `window.getComputedStyles()` but if it is attached via CSS, it doesn't show up.

I guess because it isn't actually rendering anything, it is pretty challenging to do. Ideally we would actually not render anything if `visible=False` and then we could just check the element doesn't exist but we can't do that currently for various reasons.

I think this is probably the best we can do until we make some architectural changes and/ or add some visual testing.

File: js/app/src/components/Group/Group.test.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { test, describe, assert, afterEach, vi } from "vitest";
+import { cleanup, render } from "@gradio/tootils";
+
+import Group from "./Group.svelte";
+
+describe("Group", () => {
+	afterEach(() => {
+		cleanup();
+		vi.restoreAllMocks();
```
Comment (pngwn) [inline js/app/src/components/Group/Group.test.ts:9]: We can remove this since we haven't mocked anything in this test.

```suggestion
```

**Review by pngwn — APPROVED** [review]
See the post comments for more info, I think this is okay for now. It will prevent regression and we can improve our testability at some point as well as our testing!

**Review by aliabid94 — APPROVED** [review]

**Comment by abidlabs** [comment]
Ah I see, that’s good to know ahead of the test hacking. Thanks for the review!

**Review by aliabid94 — APPROVED** [review]
Works. I hate gr.Group and gr.Box, they render so poorly, we should fix or deprecate in 4.0.

**Comment by abidlabs** [comment]
Hmm really? I never really use them myself but feel free to open an issue @aliabid94 and we can assign to 4.0

**Comment by taesiri** [comment]
@abidlabs 

I want to test this right away for my app. If I install Gradio from sources, should this work? I tested Gradio version `3.35.2` (installed from GitHub 3 minutes ago), but I can't hide Groups by setting `visible=False`.

@abidlabs 

> Works. I hate gr.Group and gr.Box, they render so poorly, we should fix or deprecate in 4.0.

Please don't 😢!  Groups are actually useful for my tasks.

**Comment by pngwn** [comment]
@taesiri Yes. If you install gradio from the `main` branch (and build the frontend), you should be able to test this change.

**Comment by abidlabs** [comment]
Yes @taesiri you can install from `main` by following the instructions here: https://gradio.app/docs/main/

**Comment by taesiri** [comment]
@pngwn  @abidlabs Thank you both. I just use the latest commit and it worked!

**Comment by apolinario** [comment]
This issue seems to be back on `3.36.1`

**Pull request merged**